internal/db/fetcher/document.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package fetcher
+
+import (
+	"bytes"
+	"context"
+
+	dsq "github.com/ipfs/go-datastore/query"
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore/iterable"
+	"github.com/sourcenetwork/defradb/internal/db/base"
+	"github.com/sourcenetwork/defradb/internal/keys"
+)
+
+// document is the type responsible for fetching documents from the datastore.
```

Comment:
question/suggestion:  Is this only for active documents? or can work for both?

Perhaps this suggestion if only for active:

```suggestion
// document is the type responsible for fetching only active documents from the datastore.
```